test/test_profiler.py

```diff
@@ -1127,6 +1127,28 @@ def test_utils_compute_self_time(self):
                     for child in event_key.event.children
                 ]))
 
+    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is required")
```

Comment:
This doesn't really test the algorithm; it just checks that it runs and produces values that are in bounds. I would suggest that you break it into the following two tests:

A real test with very coarsely controlled timings. The "real" test should be something like:
```
x = torch.ones((4096, 4096), device="cuda")
for _ in range(5):
  y = torch.mm(x, x)
torch.cuda.synchronize()
for _ in range(3):
  x[0] += 1
  time.sleep(0.1)
```
We want a region well into the pipelined queue domain and one well into the dispatch bound domain. The code above is just off the top of my head, so you may need to tune it. You should then be able to write more quantitative tests than just `>= 0.0` and `<= 1.0`

A test where you mock `_KinetoEvent` and assert that the algorithm produces exact values.